### ISSUE TITLE
No colors when publishing from Windows Powershell

### ISSUE THREAD

**Issue by RaymondLoranger** [hexpm/hex#369]
When I publish (mix hex.publish), I get these lines:
  Building docs...
  [32mDocs successfully generated.[0m
  [32mView them at "doc/index.html".[0m
instead of getting the above 2 lines in green.

If in iex, I type: IO.puts "\e[32mDocs successfully generated.\e[0m"
it works: Docs successfully generated does print in green.

**Comment by RaymondLoranger** [comment]
I am using 1.4.2. So it is just a matter of waiting for the next version?

**Comment by josevalim** [comment]
The bug was on v1.4.0 only so it is supposedly fixed on v1.4.2.

**Comment by OvermindDL1** [comment]
For note, colors work for me, but I'm using the conemu terminal in front of powershell.  What terminal is @RaymondLoranger using in front of powershell?

**Comment by RaymondLoranger** [comment]
Thanks OvermindDL1. I am using PowerShell, **not** ConEmu. (I sure intend to install it though as it looks awesome). However, the fact remains that when I run `mix docs` locally I do get the 2 lines above in green. Not so, when I run `mix hex.publish`. Hex does not use IO.ANSI. To get the passphrase, it does: `IO.write(:standard_error, "\e[2K\r#{prompt} ")`. Just a hunch, but I suspect that not including a reset (\e[0m) as part of the prompt (like IO.ANSI.format would do) messes up the console. Other than that colors work beautifully in PowerShell.

**Comment by OvermindDL1** [comment]
> I am using PowerShell, not ConEmu.

Powershell is a shell, not a terminal.  Powershell can be used in `cmd` (the default windows terminal, which is crap, but *does* support colors fine as of Windows 10, just not before), or in conemu (supports it all fine, and tabs, and more and more), and you can even use powershell in a GUI mode (has a cool debugger then too), so the 'powershell' part is not what shows color, but rather what displays is.  :-)

If you are on pre-windows10 and using 'cmd' as the terminal to powershell then colors work only if done in certain ways (I.E. not really).  If on Windows 10 and using 'cmd' then colors should work fine.  If on anything and using basically any other terminal with powershell then colors should work anyway.  :-)

So basically, I'm guessing you are using something older than Windows 10 and using the Windows built-in 'cmd' terminal (the ugly black window with white text)?  :-)
If you update to Windows 10, or use any other terminal, then it should work fine.  :-)

However yes, IEX does definitely need to be using IO.ANSI, writing the codes straight will fail in many many cases...

My terminal prompt on Windows 10:
![image](https://cloud.githubusercontent.com/assets/401605/25863953/e150d08a-34aa-11e7-8b73-0850607c449e.png)

**Comment by RaymondLoranger** [comment]
As I said, colors work beautifully using Windows PowerShell:
![image](https://cloud.githubusercontent.com/assets/4298228/25865354/9b683a8e-34c0-11e7-8658-58dc9d37b8a7.png)
Colors do not work using Windows Command Prompt:
![image](https://cloud.githubusercontent.com/assets/4298228/25865463/f2ccbcbe-34c0-11e7-846d-2944b6e1200e.png)
I am on Windows 10.

**Comment by josevalim** [comment]
What happens if you run this on IEx:

    IO.puts :stderr, IO.ANSI.format [:green, "foo"]

**Comment by OvermindDL1** [comment]
This is what I get in the Windows 10 stock `cmd` prompt:
![image](https://cloud.githubusercontent.com/assets/401605/25866296/3401052c-34b3-11e7-9c38-b6810ad58f2b.png)

And in the Win10 stock powershell prompt:
![image](https://cloud.githubusercontent.com/assets/401605/25866349/6fae3f04-34b3-11e7-8265-f9c9546c8112.png)

**Comment by OvermindDL1** [comment]
And for completion, in my mingw shell in conemu:
![image](https://cloud.githubusercontent.com/assets/401605/25866444/c3fab02e-34b3-11e7-8881-688dacd9d884.png)

**Comment by RaymondLoranger** [comment]
I have to explicitly specify `true` to get green:
![image](https://cloud.githubusercontent.com/assets/4298228/25866843/cf4c4278-34c5-11e7-9416-c97124650da5.png)
My .iex.exs has this configuration:
IEx.configure colors: [
  enabled: true,
  eval_result: [:cyan, :bright],
  eval_info: [:blue, :bright],
  eval_error: [:magenta, :bright],
  doc_title: [:reverse, :yellow, :bright],
  doc_headings: [:yellow, :bright],
  doc_inline_code: [:magenta, :bright],
  ls_directory: [:green, :bright],
  stack_app: [:green, :bright],
  stack_info: [:red, :bright]
]

**Comment by RaymondLoranger** [comment]
the color issue when publishing in Powershell is now resolved (whatever the explanation may be):
![image](https://cloud.githubusercontent.com/assets/4298228/26414962/e23624c0-407e-11e7-9b82-68a51fdc3396.png)

**Comment by OvermindDL1** [comment]
> the color issue when publishing in Powershell is now resolved (whatever the explanation may be):

Huh, was elixir updated?  Hex updated?  Windows updated?  Powershell updated?

**Comment by RaymondLoranger** [comment]
On my part, I installed Elixir v1.4.4 (before I had Elixir v1.4.2).  And yes ConEmu would give me colors prior to v1.4.4.  Thanks much.

**Comment by RaymondLoranger** [comment]
I forgot to mention that I also recently installed the Windows 10 Creators Update.  That could be the explanation.
